### PR TITLE
sdk_min_versions plugin

### PR DESCRIPTION
**Pull request by davidbielik**
A plugin for listing SDK minimum version requirements

* Specify each platform and corresponding version
* Automatically links to the changelog with version anchor
* Looks really cool
* Thanks @radixdev for doing all the hard work

usage `{% sdk_min_versions ios:3.2.3 android:8.0.0 web:1.2.3 %}`

image

### CODE REVIEW

File: _plugins/sdk_min_versions_tag.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      end
+
+      def render(context)
+        puts "doing render"
+        site = context.registers[:site]
+        converter = site.find_converter_instance(::Jekyll::Converters::Markdown)
+
+        render_html = "<div id='sdk-versions'>"
+        if !@original_web.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk'>Web: #{@original_web}+</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:29]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk'>Web: #{@original_web}</a>"
```

File: _plugins/sdk_min_versions_tag.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        end
+        if !@ios_changelog_ref.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/ios/changelog/##{@ios_changelog_ref}' class='sdk-versions--chip ios-sdk'>iOS: #{@original_ios}+</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:32]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/ios/changelog/##{@ios_changelog_ref}' class='sdk-versions--chip ios-sdk'>iOS: #{@original_ios}</a>"
```

File: _plugins/sdk_min_versions_tag.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        end
+        if !@android_changelog_ref.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/android/changelog/##{@android_changelog_ref}' class='sdk-versions--chip android-sdk'>Android: #{@original_android}+</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:35]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/android/changelog/##{@android_changelog_ref}' class='sdk-versions--chip android-sdk'>Android: #{@original_android}</a>"
```

File: assets/css/sdk_min_versions.css [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    margin-bottom: 30px;
+}
+
+#sdk-versions .sdk-versions--chip {
+    word-break: normal;
+    tab-size: 4;
+    text-rendering: optimizeLegibility;
+    -webkit-font-smoothing: antialiased;
+    box-sizing: inherit;
+    margin: 0;
```
Comment (davidbielik) [inline assets/css/sdk_min_versions.css:11]:
```suggestion
    margin: 0 4px;
```

File: _plugins/sdk_min_versions_tag.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      end
+
+      def render(context)
+        puts "doing render"
+        site = context.registers[:site]
+        converter = site.find_converter_instance(::Jekyll::Converters::Markdown)
+
+        render_html = "<div id='sdk-versions'>"
+        if !@original_web.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk'>Web: #{@original_web}</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:29]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk' target="_blank">Web: #{@original_web}</a>"
```

File: _plugins/sdk_min_versions_tag.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        end
+        if !@ios_changelog_ref.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/ios/changelog/##{@ios_changelog_ref}' class='sdk-versions--chip ios-sdk'>iOS: #{@original_ios}</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:32]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/ios/changelog/##{@ios_changelog_ref}' class='sdk-versions--chip ios-sdk' target="_blank">iOS: #{@original_ios}</a>"
```

File: _plugins/sdk_min_versions_tag.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        end
+        if !@android_changelog_ref.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/android/changelog/##{@android_changelog_ref}' class='sdk-versions--chip android-sdk'>Android: #{@original_android}</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:35]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/android/changelog/##{@android_changelog_ref}' class='sdk-versions--chip android-sdk' target="_blank">Android: #{@original_android}</a>"
```

File: _plugins/sdk_min_versions_tag.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      end
+
+      def render(context)
+        puts "doing render"
+        site = context.registers[:site]
+        converter = site.find_converter_instance(::Jekyll::Converters::Markdown)
+
+        render_html = "<div id='sdk-versions'>"
+        if !@original_web.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk' target="_blank">Web: #{@original_web}</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:29]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk' target='_blank'>Web: #{@original_web}</a>"
```

File: _plugins/sdk_min_versions_tag.rb [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        end
+        if !@ios_changelog_ref.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/ios/changelog/##{@ios_changelog_ref}' class='sdk-versions--chip ios-sdk' target="_blank">iOS: #{@original_ios}</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:32]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/ios/changelog/##{@ios_changelog_ref}' class='sdk-versions--chip ios-sdk' target='_blank'>iOS: #{@original_ios}</a>"
```

File: _includes/html_include.html [hunk 10]
```diff
@@ -29,6 +29,8 @@
 
 <link type="text/css" rel="stylesheet" href="{{site.baseurl}}/assets/css/swiper.css"/>
 
+<link type="text/css" rel="stylesheet" href="{{site.baseurl}}/assets/css/sdk_min_versions.css" />
```
Comment (zzhaobraze) [inline _includes/html_include.html:32]: you can probably add this to the https://github.com/braze-inc/braze-docs/blob/develop/assets/css/application.scss and included it as part of the `scss` load

File: _plugins/sdk_min_versions_tag.rb [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        end
+        if !@android_changelog_ref.nil?
+          render_html += "<a href='/developer_guide/platform_integration_guides/android/changelog/##{@android_changelog_ref}' class='sdk-versions--chip android-sdk' target="_blank">Android: #{@original_android}</a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:35]:
```suggestion
          render_html += "<a href='/developer_guide/platform_integration_guides/android/changelog/##{@android_changelog_ref}' class='sdk-versions--chip android-sdk' target='_blank'>Android: #{@original_android}</a>"
```

**Comment by davidbielik** [comment]
Thanks @zzhaobraze i moved the CSS into the `scss` pattern, and QA'd it in another branch i'm working on.

File: _plugins/sdk_min_versions_tag.rb [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+        @original_web = get_full_version(text, 'web')
+        @web_changelog_ref = get_changelog_ref(@original_web)
+
+        @original_android = get_full_version(text, 'android')
+        @android_changelog_ref = get_changelog_ref(@original_android)
+      end
+
+      def render(context)
+        puts "doing render"
```
Comment (zzhaobraze) [inline _plugins/sdk_min_versions_tag.rb:23]: @davidbielik think you left this from debugging

Comment (davidbielik) [reply _plugins/sdk_min_versions_tag.rb]: thanks! removed

File: _plugins/sdk_min_versions_tag.rb [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        @android_changelog_ref = get_changelog_ref(@original_android)
+      end
+
+      def render(context)
+        site = context.registers[:site]
+        converter = site.find_converter_instance(::Jekyll::Converters::Markdown)
+
+        render_html = "<div id='sdk-versions'>"
+        if !@original_web.nil?
+          render_html += "<a href='/docs/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk' target='_blank'><i class="fa fa-desktop"></i> &nbsp; Web: #{@original_web} &nbsp;<i class='fa fa-external-link'></i></a>"
```
Comment (davidbielik) [inline _plugins/sdk_min_versions_tag.rb:28]:
```suggestion
          render_html += "<a href='/docs/developer_guide/platform_integration_guides/web/changelog/##{@web_changelog_ref}' class='sdk-versions--chip web-sdk' target='_blank'><i class='fa fa-desktop'></i> &nbsp; Web: #{@original_web} &nbsp;<i class='fa fa-external-link'></i></a>"
```

**Comment by zzhaobraze** [comment]
@davidbielik I'm fine with including other custom liquid tags but I believe there might be an bug when the tag is missing parameters. For example:
`{% sdk_min_versions  %} renders the ios and android tag`
image

and removing various combination still shows the platform
`{% sdk_min_versions android:8.0 web:1.2.3 %}` (iOS without a version)
image

**Review by zzhaobraze — APPROVED** [review]

**Comment by zzhaobraze** [comment]
@KellieHawks @timothy-kim feel free to merge whenever you feel comfortable with it.

**Pull request merged**